src/FindEmailAddress.java

```diff
@@ -0,0 +1,362 @@
+/*
+*
+* The project is for finding email address in the web site
+*
+* Create a command line program that will take an internet domain name (i.e. “jana.com”) and
+* print out a list of the email addresses that were found on that website only.
+* the application will find email addresses on any discoverable page of the website.
+*
+* We implemented a multithread method. each url request and email&href check is a threadTask
+*
+*
+*/
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Hashtable;
+import java.net.URI;
+import java.util.concurrent.*;
+
+/*
+*  This is the main class to get the email addresses
+*
+*/
+public class FindEmailAddress {
+
+    private String rootURL;
+    public String requestDomain;
+    public Hashtable<String,String> urlHistory;
+    public Hashtable<String,String> emails;
+    public static int maxUrlCheck = 200; //to avoid too many pages to check we add maximum number of url to check, it can be modified
+    public ExecutorService executor;
+    public Integer tasksNumber;
+
+    public static void main(String[] args)
+    {
+        String requestUrl = null;
+
+        if(args.length >0)
+        {
+            requestUrl =  args[0];
+        }
+        if (requestUrl == null)
+            return;
```

Comment:
exiting cleanly here with no error message and status 0 is strange; I would have printed a usage string.

Reply:
Good Suggestion.